### PR TITLE
Use smaller integer field types for GTFS entity classes

### PR DESCRIPTION
**Pull request by aababilov**
GTFS enums usually fit into one byte, so we don't need an int. GTFS tables usually less than 16 fields, so we need a byte or a short int for storing a bitmask of assigned fields.

Smaller fields allow us to save memory for the largest table - stop_times.txt that may have 30 M lines.

Instead of:
```
class GtfsStopTime {
  // ...
  private int pickupType;
  private int dropOffType;
  private int continuousPickup;
  private int continuousDropOff;
  private int timepoint;
  private int bitField0_;
}
```
we have now:
```
class GtfsStopTime {
  // ...
  private byte pickupType;
  private byte dropOffType;
  private byte continuousPickup;
  private byte continuousDropOff;
  private byte timepoint;
  private short bitField0_;
}
```
which is 5 * (4 - 1) + 2 = 17 bytes smaller. Java aligns classes by 8 bytes, so we actually save 16 bytes per line.

Total save: 0.5 GiB for 30 M lines in stop_times.txt

### CODE REVIEW

File: processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/EntityImplementationGenerator.java [hunk 1]
```diff
@@ -59,14 +62,44 @@ private enum ClassContext {
...
+    }
+    if (minValue >= Short.MIN_VALUE && maxValue <= Short.MAX_VALUE) {
+      return TypeName.SHORT;
+    }
+    return TypeName.INT;
   }
 
   private static int lastBitFieldNumber(int fieldCount) {
-    // Each bitField has 32 bits. We need bitField0_ to store 1..32 fields,
+    // All standard GTFS tables have less than 16 fields, so a bitmask fits into short.
```
Comment (asvechnikov2) [inline processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/EntityImplementationGenerator.java:100]: The comment "All standard GTFS" will become obsolete relatively soon, what do you think about replacing it with "Most standard GTFS"?

File: processor/tests/src/test/java/org/mobilitydata/gtfsvalidator/processor/tests/EnumSizesSchemaTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    assertThat(enumSizes.byteEnum()).isEqualTo(ByteGtfs.UNRECOGNIZED);
+    // However, 10 can be still retrieved using byteEnumValue method.
+    assertThat(enumSizes.byteEnumValue()).isEqualTo(10);
+  }
+
+  @Test
+  public void testByteEnumOutOfBounds() {
+    // Short.MIN_VALUE is out of bounds for byte. It cannot be retrieved with byteEnumValue.
+    assertThat(new EnumSizes.Builder().setByteEnum((int) Short.MIN_VALUE).build().byteEnumValue())
+        .isEqualTo(0);
```
Comment (asvechnikov2) [inline processor/tests/src/test/java/org/mobilitydata/gtfsvalidator/processor/tests/EnumSizesSchemaTest.java:39]: Should we throw any exception if values can't be retrieved? It looks like an undefined behaviour

File: processor/tests/src/main/java/org/mobilitydata/gtfsvalidator/processor/tests/ManyFieldsSchema.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package org.mobilitydata.gtfsvalidator.processor.tests;
+
+import org.mobilitydata.gtfsvalidator.annotation.GtfsTable;
+
+@GtfsTable("many_fields.txt")
+public interface ManyFieldsSchema {
```
Comment (asvechnikov2) [inline processor/tests/src/main/java/org/mobilitydata/gtfsvalidator/processor/tests/ManyFieldsSchema.java:6]: Do you think it's worth testing an edge case when the number of fields is equal to the number of bits and the last field is handled correctly?

**Review by asvechnikov2 — COMMENTED** [review]
Thanks for the tests!

**Comment by f8full** [comment]
Nice

File: processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/EntityImplementationGenerator.java (hunk 1 above)
Comment (aababilov) [reply processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/EntityImplementationGenerator.java:98]: Done.

File: processor/tests/src/test/java/org/mobilitydata/gtfsvalidator/processor/tests/EnumSizesSchemaTest.java (hunk 2 above)
Comment (aababilov) [reply processor/tests/src/test/java/org/mobilitydata/gtfsvalidator/processor/tests/EnumSizesSchemaTest.java]: We generate UnexpectedEnumValueNotice and store UNRECOGNIZED.

File: processor/tests/src/main/java/org/mobilitydata/gtfsvalidator/processor/tests/ManyFieldsSchema.java (hunk 3 above)
Comment (aababilov) [reply processor/tests/src/main/java/org/mobilitydata/gtfsvalidator/processor/tests/ManyFieldsSchema.java:6]: Changed to 8 fields.

**Comment by aababilov** [comment]
> Nice

Thanks!

**Comment by aababilov** [comment]
> wave Welcome back!

I missed GTFS Validator :)

**Review by asvechnikov2 — APPROVED** [review]
LGTM! Thanks!
